src/vms/pvm/etna-builder/spendHelper.ts

```diff
@@ -0,0 +1,255 @@
+import type { TransferableOutput } from '../../../serializable';
+import { TransferableInput } from '../../../serializable';
+import type { Utxo } from '../../../serializable/avax/utxo';
+import { bigIntMin } from '../../../utils/bigintMath';
+import { compareTransferableOutputs } from '../../../utils/sort';
+import type { Dimensions } from '../../common/fees/dimensions';
+import { addDimensions, dimensionsToGas } from '../../common/fees/dimensions';
+import { getInputComplexity, getOutputComplexity } from '../txs/fee';
+
+interface SpendHelperProps {
+  changeOutputs: readonly TransferableOutput[];
+  complexity: Dimensions;
+  gasPrice: bigint;
+  inputs: readonly TransferableInput[];
+  stakeOutputs: readonly TransferableOutput[];
+  toBurn: Map<string, bigint>;
+  toStake: Map<string, bigint>;
+  weights: Dimensions;
+}
+
+/**
+ * The SpendHelper class assists in managing and processing the spending of assets,
+ * including handling complexities, gas prices, and various outputs and inputs.
+ *
+ * @class
+ */
+export class SpendHelper {
```

Comment:
Can we add tests for this class? It's doing a lot of heavy lifting and is being mutated, so we need to make sure the side-effects do not break any contracts with the consumers of this class

Reply:
Initial set of tests have been added. I'll adjust and add more as we go into the next draft of work to refactor the spend logic to be more idiomatic.